lib/pure/net.nim

```diff
@@ -180,6 +188,22 @@ when defined(ssl):
     var errStr = ErrErrorString(err, nil)
     raise newException(SSLError, $errStr)
 
+  proc getSslContextExtraDataIndex*(): cint =
+    ## Retrieves unique index for storing extra data in SSLContext.
+    return SSL_CTX_get_ex_new_index(0, nil, nil, nil, nil)
+
+  proc setExtraData*(ctx: SSLContext, index: cint, data: pointer) =
```

Comment:
I feel like this should be a bit more idiomatic and take a ``RootRef`` instead of a ``pointer``.

@Araq thoughts?